File: web/src/types/mailbox.ts
export interface Mailbox {
    id: number;
    created_at: string;
    updated_at: string;
    deleted_at?: string;
    name: string;
    user_name: string;
    password: string;
    tls_enabled: boolean;
    maximum_size: number;
}
interface MailMessageHeader {
    From:string
    To: string
}
export interface MailMessage {
    id: number;
    created_at: Date;
    updated_at: Date;
    sender: string;
    subject: string;
    receiver: string;
    message: string;
    mail_box_id: number;
    headers: MailMessageHeader
}

File: db/models/mail_box.go
package models

import (
	"gorm.io/gorm"
	"time"
)

type Model struct {
	ID        uint           `gorm:"primary_key" json:"id"`
	CreatedAt *time.Time     `json:"created_at,omitempty" gorm:"type:datetime"`
	UpdatedAt *time.Time     `json:"updated_at,omitempty" gorm:"type:datetime"`
	DeletedAt gorm.DeletedAt `json:"deleted_at,omitempty"  gorm:"type:datetime"`
}
type MailBox struct {
	Model
	Name        string `json:"name" gorm:"type:varchar(255)"`
	UserName    string `json:"user_name" gorm:"type:varchar(255)"`
	Password    string `json:"password" gorm:"type:varchar(255)"`
	TlsEnabled  bool   `json:"tls_enabled"`
	MaximumSize int8   `json:"maximum_size"`
}
